solr/core/src/java/org/apache/solr/response/transform/ChildDocTransformerFactory.java

```diff
@@ -106,9 +107,12 @@ public DocTransformer create(String field, SolrParams params, SolrQueryRequest r
       }
     }
 
+    String childReturnFields = params.get("fl");
+    SolrReturnFields childSolrReturnFields = childReturnFields==null? new SolrReturnFields(): new SolrReturnFields(childReturnFields ,req);
```

Comment:
Shouldn't this default to the top level "fl"?  I think so.  Even if it didn't do this before, it in-effect behaves this way and it makes sense (to me) that it would.

Reply:
That makes sense,
just pushed a commit :).